lib/buffer.js

```diff
@@ -1314,10 +1316,23 @@ function atob(input) {
   return Buffer.from(input, 'base64').toString('latin1');
 }
 
+function isUtf8(input) {
+  if (isTypedArray(input) || Buffer.isBuffer(input)) {
+    return bindingIsUtf8(input.buffer);
```

Comment:
If `input` input uses the internal `Buffer` pool, doesn't this use the whole memory pool?

Reply:
Or if input is a subarray.